src/app/profile/yourGiving/recipientView/recipientGift/recipientGift.component.js

```diff
@@ -31,22 +31,38 @@ class RecipientGift {
     if ( this.showDetails && !this.detailsLoaded ) {
       this.isLoading = true;
 
-      //get payment methods
-      this.profileService.getPaymentMethods( true )
-        .subscribe( ( paymentMethods ) => {
-          angular.forEach(this.recipient.donations, (donation) => {
-            let paymentMethodId = donation['historical-donation-line']['payment-method-id'];
-            donation['paymentmethod'] = find(paymentMethods, {id: paymentMethodId});
-          });
+      let paymentMethodRequests = [], paymentMethodUris = [];
+      angular.forEach(this.recipient.donations, (donation) => {
+        let uri = donation['payment-method-link'] && donation['payment-method-link']['uri'];
+        if(uri && paymentMethodUris.indexOf(uri) < 0){
+          paymentMethodUris.push(uri);
+          paymentMethodRequests.push(this.profileService.getPaymentMethod( uri, true ));
+        }
+      });
+
+      if(paymentMethodRequests.length){
+        Observable.forkJoin(paymentMethodRequests)
+          .subscribe((response) => {
+            angular.forEach(response, (paymentMethod) => {
```

Comment:
You could avoid a nested loop if you use [keyBy](https://lodash.com/docs/4.17.4#keyBy) on the payment methods first and then for each donation, just look that value up in the object keyBy created.